system-test/error-reporting.ts

```diff
@@ -163,18 +165,18 @@ describe('Request/Response lifecycle mocking', function() {
         new Configuration({ignoreEnvironmentCheck: true}, logger), logger);
   });
 
-  afterEach(function() {
+  afterEach(() => {
     env.sterilizeProcess();
     nock.cleanAll();
   });
 
-  after(function() {
+  after(() => {
     env.restoreProcessToOriginalState();
   });
 
-  it('Should fail when receiving non-retryable errors', function(done) {
+  it('Should fail when receiving non-retryable errors', function(this, done) {
```

Comment:
You can also just hang the `.timeout(...)`  off of the end of the method:  
```js
it('should be faster I guess', () => {
  // do tests
}).timeout(100000000);
```

Reply:
Oh cool, I didn't know about that.

Reply:
We talked about it, and I decided to use the form
```ts
function(this, done) {
  this.setTimeout(42);
  ...
}
```
since I feel its easier to read the test because at the start of the test you know that it has a unique timing expected.

We can always change this if needed in a follow-up PR.